docs/rfcs/0004-rfc-connectiomanager.md

```diff
@@ -0,0 +1,74 @@
+- Status: In progress
+- Date: 2023-11-02
+- Author: [@gela](https://github.com/gela)
+- RFC PR: TBD
+- Implementation GitHub issue: TBD.
+
+
+## Summary
+
+The existing `ConnectionManager` class will be renamed to `ConnectionDiagnostics` and will be responsible for handling all connectivity related checks once during VPN activation (`Controller::StateConnecting`) and then periodically once the VPN is already activated (`Controller::StateOn`); the controller will be the entity which manages the controller states that the client goes through such as `connecting` and `confirming` as we activate and deactivate the VPN to reflect those to the user what is happening behind the scene.
```

Comment:
Why do we need to run diagnostics while the VPN is just working as expected? I'd expect that to be necessary only once ConnectionHealth has reported instability or lack of signal.

Reply:
Actually reading on it seems we only run diagnostics when we go into NoSignal. All good.